### ISSUE TITLE
SLPO’s moment-matched Gaussian is a surrogate score, not the dropout policy density

### ISSUE THREAD

**Issue by microprediction** [microprediction/latentreasoning#71]
## Scope reviewed

Current default-branch commit: [11b700ba0c9cc03b0e1a54db15fe06d15fad8e51](https://github.com/microprediction/latentreasoning/commit/11b700ba0c9cc03b0e1a54db15fe06d15fad8e51).

Affected text:

- [training.html lines 239–242](https://github.com/microprediction/latentreasoning/blob/11b700ba0c9cc03b0e1a54db15fe06d15fad8e51/training.html#L239-L242), which says every listed method adds noise in a space that “fixes the form of the log-probability”
- [training.html lines 282–286](https://github.com/microprediction/latentreasoning/blob/11b700ba0c9cc03b0e1a54db15fe06d15fad8e51/training.html#L282-L286), which groups SLPO with ordinary GRPO/RLOO estimators
- [training.html lines 410–418](https://github.com/microprediction/latentreasoning/blob/11b700ba0c9cc03b0e1a54db15fe06d15fad8e51/training.html#L410-L418), especially “SLPO uses the same source of randomness to build a density rather than a sample”

I searched open and closed issues for SLPO, surrogate likelihood, Gaussian score, and score alignment and found no existing issue covering this distinction.

## Defect

SLPO does **not** recover the rollout policy density from dropout. Its behavior trajectories are generated by MC dropout, but it fits a moment-matched isotropic Gaussian and uses that Gaussian only as a **surrogate score**.

The paper is explicit:

- [Section 4.2](https://arxiv.org/html/2607.19691#S4.SS2) says MC dropout supplies behavior trajectories while the Gaussian “converts their sampled latent states into a differentiable score.”
- The realized latent state is a stop-gradient target; gradients run through recomputed Gaussian moments, not through the rollout law.
- [Appendix A.5, Proposition 2](https://arxiv.org/html/2607.19691#A5.Thmproposition2) establishes only that the implementation optimizes a detached empirical surrogate objective.
- [Proposition 3](https://arxiv.org/html/2607.19691#A5.Thmproposition3) approximates the expected-reward gradient only under an explicit bound on the mismatch between the surrogate score and the true behavior score.
- [Proposition 4](https://arxiv.org/html/2607.19691#A5.Thmproposition4) gives reward ascent only when that approximation error is small enough relative to the surrogate direction.

This is not a semantic nicety. For a scalar toy dropout transition

~~~text
h = m * theta,   m ~ Bernoulli(1/2),
~~~

the behavior law is the two-atom distribution `q_theta = 0.5 delta_0 + 0.5 delta_theta`. Its moment-matched Gaussian is continuous and assigns positive density away from both atoms. Therefore the Gaussian log-score is not `log q_theta(h)`, and its gradient is not automatically the policy score. The same support mismatch remains in a finite neural network with Bernoulli dropout, however many masks it has.

## Consequence

The page currently makes SLPO look like the same construction as methods with an explicit Gaussian transition policy, such as CoLaR. That overstates the estimator’s status:

- RLOO/GRPO supplies SLPO’s advantage normalization, but does not turn the Gaussian surrogate into the behavior-policy likelihood.
- Empirical gains support the method, but the update is not generally an unbiased score-function estimator of the dropout rollout objective.
- The paper’s score-alignment and local-ascent conditions are the exact qualifications needed to connect the surrogate update to expected reward.

## Suggested correction

Describe the SLPO row separately from exact transition-density methods. For example:

> MC dropout generates behavior trajectories. Repeated dropout evaluations parameterize a moment-matched isotropic Gaussian surrogate, and the sampled latent is treated as a stop-gradient target while gradients flow through the surrogate moments. RLOO or GRPO supplies the advantage. This optimizes a detached empirical surrogate objective; agreement with the expected-reward policy gradient requires the paper’s score-alignment condition.

The reported benchmark gains and adaptive stopping result need not change. The correction is to distinguish an exact action likelihood from a learned/moment-matched surrogate score.

**Comment by microprediction** [comment]
Correct, fixed in 1185867.

Verified: section 4.2 says the Gaussian "converts their sampled latent states into a differentiable score", the realized latent is a stop-gradient target, and the gradient flows through recomputed moments rather than the rollout law. Propositions 2 through 4 exist precisely to bound when that surrogate gradient tracks the true one, under a score-mismatch condition.

The page had SLPO grouped with methods whose noise fixes the form of the log-probability exactly, which elides the thing the paper's own title flags. It now says the Gaussian is a surrogate fitted to dropout samples rather than the dropout law itself, and that the paper carries bounds on when its gradient tracks the real one.
